File: Java/GithubApiSample/app/src/main/java/com/example/githubapisample/network/ServerApi.java
package com.example.githubapisample.network;

import com.example.githubapisample.model.User;
import com.example.githubapisample.model.UserList;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Path;

public interface ServerApi {

    @GET("users")
    Call<List<UserList>> getUsers(
            @Header("Authorization") String token
    );

    @GET("users/{login}")
    Call<User> getUserInfo(
            @Header("Authorization") String token,
            @Path("login") String login
    );

}

File: Kotlin/ViewModel/app/src/main/java/com/example/viewmodel/ServerApi.kt
package com.example.viewmodel

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface ServerApi {

    @GET("posts/1")
    fun getPost() : Call<Post>

    @GET("posts/{number}")
    fun getPost2(
        @Path("number") number : Int
    ) : Call<Post>

}